src/niscope/system_tests/test_system_niscope.py

```diff
@@ -66,6 +67,23 @@ def test_fetch_binary8_into(session):
     assert len(wfm_infos) == test_num_channels
 
 
+def test_fetch_binary8(session):
+    test_voltage = 1.0
+    test_record_length = 2000
+    test_channels = range(2)
+    test_num_channels = 2
+    wfm = numpy.ndarray(test_num_channels * test_record_length, dtype=numpy.int8)
+    # Initialize with NaN so we can later verify all samples were overwritten by the driver.
+    wfm.fill(float('nan'))
+    session.configure_vertical(test_voltage, niscope.VerticalCoupling.AC)
+    session.configure_horizontal_timing(50000000, test_record_length, 50.0, 1, True)
+    with session.initiate():
+        wfm_infos = session.channels[test_channels].fetch(timeout=5.0, wfm=wfm)
```

Comment:
> If wfm is not None calls the appropriate fetch_into() function

I thought we had decided against doing that kind of stuff ie having functions that both mutate their parameter and return it?

Reply:
It doesn't directly return the wfm. It puts slices/memoryviews of it into the wfm_info object.